library/src/com/google/maps/android/kml/KmlStyle.java

```diff
@@ -0,0 +1,369 @@
+package com.google.maps.android.kml;
+
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.maps.model.PolygonOptions;
+import com.google.android.gms.maps.model.PolylineOptions;
+
+import android.graphics.Color;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * Represents the defined styles in the KML document
+ */
+public class KmlStyle {
+
+    private final static int HSV_VALUES = 3;
+
+    private final static int HUE_VALUE = 0;
+
+    private final static int INITIAL_SCALE = 1;
+
+    private final MarkerOptions mMarkerOptions;
+
+    private final PolylineOptions mPolylineOptions;
+
+    private final PolygonOptions mPolygonOptions;
+
+    private final HashMap<String, String> mBalloonOptions;
+
+    private final HashSet<String> mStylesSet;
+
+    private boolean mFill = true;
+
+    private boolean mOutline = true;
+
+    private String mIconUrl;
+
+    private double mScale;
+
+    private String mStyleId;
+
+    private boolean mIconRandomColorMode;
+
+    private boolean mLineRandomColorMode;
+
+    private boolean mPolyRandomColorMode;
+
+    private int mMarkerColor;
+
+
+    /**
+     * Creates a new Style object
+     */
+    public KmlStyle() {
+        mStyleId = null;
+        mMarkerOptions = new MarkerOptions();
+        mPolylineOptions = new PolylineOptions();
+        mPolygonOptions = new PolygonOptions();
+        mBalloonOptions = new HashMap<String, String>();
+        mStylesSet = new HashSet<String>();
+        mScale = INITIAL_SCALE;
+        mMarkerColor = Color.parseColor("#ffffff");
+        mIconRandomColorMode = false;
+        mLineRandomColorMode = false;
+        mPolyRandomColorMode = false;
+    }
+
+    /**
+     * Sets the text for the info window; no other ballonstyles are supported
+     *
+     * @param text text to put in an info window
+     */
+    public void setInfoWindowText(String text) {
+        mBalloonOptions.put("text", text);
+    }
+
+    /**
+     * @return the string representing a style id, null otherwise
+     */
+    public String getStyleId() {
+        return mStyleId;
+    }
+
+    /**
+     * @param styleId style id for this style
+     */
+    public void setStyleId(String styleId) {
+        mStyleId = styleId;
+    }
+
+    /**
+     * Checks if a given style has been set
+     *
+     * @param style style to check if set
+     * @return true if style was set, false otherwise
+     */
+    public boolean isStyleSet(String style) {
```

Comment:
I don't really like using a hashmap for this and passing in strings.  A couple of other possible options:
 - Store the style options as boxed values (e.g., Double instead of double) and use null to indicate not set.
 - Store a separate boolean for each indicating whether they have been set and add isXxxSet() methods.
 - Initialize each variable with a default value (probably the same as what the API uses) then don't worry about whether they've been explicitly set.